### DISCUSSION TITLE
Eco mode

### DISCUSSION THREAD

**Discussion by razierklinge** [RobertD502/home-assistant-iocare#41] [General]
Would it be possible to expose Eco mode as something that can be monitored and used as a trigger? This is with a 400s.

My goal is to essentially not allow Eco mode to activate. It doesnt seem like this is a setting that can be changed normally, so I would like to make an automation that detects Eco mode and then just sets it back to Smart mode. It also doesn't look like there's an action to set the fan to specific levels or Smart mode, unsure if that can be done as well. 

I'm extremely new to Home Assistant so I apologize if all of this is already possible by adding it myself somehow.

**Comment by RobertD502** [comment]
From Home Assistant's point of view, I will not be able to add Eco as a preset mode to the fan entity - only preset modes that should be exposed are those that a user can set the entity to (which Smart Eco mode isn't for the 300s/400s).

Is there any particular reason you are against Eco mode? In Smart Eco mode, the fan is shut off to conserve energy (it'll also conserve the blower motor). Once the particle sensor notices that the air quality has dropped, it will automatically kick itself out of Smart Eco mode and into normal Smart mode.

**Reply by RobertD502** [reply]
Actually I may have a way around this so users don't think they can select that mode.

**Comment by razierklinge** [comment]
Basically it seems like the sensor doesn't work _that_ well. It'll go into Eco mode even when I've cooked or cleaned a litterbox and there's obviously stuff in the air I'd like filtered. So I always want it at least running at level 1 in smart mode. I've set the smart sensitivity to max to see if that helps as well.

**Reply by RobertD502** [reply]
I have a workaround that will only show the Auto (Eco) mode (plus regular Auto and night presets) as a preset option when the device is in that mode and only Auto or Night mode options if it is in either of those modes or at a user-defined speed. An error message will be presented to users if they attempt to manually set the purifier to Auto (Eco) mode since that isn't possible. The change will be included in the next release!

Hope that helps.

**Reply by razierklinge** [reply]
Awesome! Thanks for taking the time to look into this.

**Reply by RobertD502** [reply]
@razierklinge Update to 0.4.2 and you will be able to use the preset `Auto (Eco)` as a trigger in your automation(s). This preset will only be present/the current selected preset mode when the purifier has decided to set itself into Eco mode.
